debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -131,7 +165,39 @@ public static long contiguousHeaderSize()
 		return J9IndexableObjectContiguous.SIZEOF;
 	}
 
+	/**
+	 * Get the header size of the contiguous indexable object with the dataAddress header field.
+	 */
+	public static long contiguousHeaderSizeWithDataAddress() {
+		if (mixedReferenceMode) {
+			if (compressObjectReferences) {
+				return J9IndexableObjectWithDataAddressContiguousCompressed.SIZEOF;
+			}
+			return J9IndexableObjectWithDataAddressContiguousFull.SIZEOF;
+		}
+		return J9IndexableObjectWithDataAddressContiguous.SIZEOF;
```

Comment:
I expect this to be problematic in 32-bit builds where `J9IndexableObjectWithDataAddressContiguous` is not declared.

Reply:
Correct, but shouldn't this code be unreachable in old cores as uses of `J9IndexableObjectWithDataAddress*` are protected by the `NoSuchFieldException` of `isIndexableDataAddrPresent` which has also been introduced in this commit?

Reply:
I can introduce `contiguousHeaderSize` and `discontiguousHeaderSize` variables to `J9IndexableObjectHelper.java` if this is what you are suggesting. Where in the DDR code would be the best suitable place to set these (as they must be initialized before the first use of `*contiguousHeaderSize()` calls)?

Reply:
They can be static fields of this class, lazily initialized via calls to methods here.

Reply:
Okay, this is close to what I implemented yesterday. I have just done the initialization in the `GCArrayletObjectModelBase` constructor. Will push all changes shortly.